### PR TITLE
Update upgrade tutorial

### PR DESCRIPTION
**Pull request by ipopescu**
### Related links

No story was opened for this. Discussed with Karan and decided to make the update. 

### Changes

Recommendation to use `new_contract` instead of `create_contract_package_at_hash` when creating a contract in the upgrade tutorial.

### Notes
Tech review: Michał or Karan. Writing review: Adam
Ran locally without issues.

### CODE REVIEW

File: source/docs/casper/dapp-dev-guide/tutorials/upgrade-contract.md [hunk 1]
```diff
@@ -15,29 +15,37 @@ This tutorial examines how to upgrade an existing contract, a process similar to
...
+### Step 1. Create a new unlocked contract
 
-Create a new contract package using the [create_contract_package_at_hash](https://docs.rs/casper-contract/latest/casper_contract/contract_api/storage/fn.create_contract_package_at_hash.html#) function to store the new (versioned) contract under a key in global state.
+Create a new contract using the [new_contract](https://docs.rs/casper-contract/latest/casper_contract/contract_api/storage/fn.new_contract.html) function and store the ContractHash returned under a key in global state for later access. Under the hood, the execution engine will create a contract package (a container for the contract) that can be versioned.
 
-You need the access key URef returned from `create_contract_package_at_hash` in your context, giving you access to your named keys. Without the access key URef, you cannot add new contract versions for security reasons.
+When creating the contract, you can specify the package name and access URef for further modifications. Without the access key URef, you cannot add new contract versions for security reasons. Optionally, you can also save the latest version of the contract package under a named key.
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/tutorials/upgrade-contract.md:29]: @mpapierski or @darthsiroftardis, if we store the ContractHash for later modifications, is the access key URef optional? This is what I understand when I read [this](https://docs.rs/casper-contract/latest/casper_contract/contract_api/storage/fn.new_contract.html).

**Comment by ipopescu** [comment]
@ACStoneCL or @ramujin could you please review? ideally, this documentation goes live before Davos on Monday. Thanks!

**Review by ACStoneCL — APPROVED** [review]
Changes look good to me, just one potential word usage suggestion.

**Comment by ipopescu** [comment]
Thanks, @ACStoneCL!

**Pull request merged**